cfme/tests/cli/test_appliance_console.py

```diff
@@ -269,81 +276,88 @@ def is_storage_extended():
     wait_for(is_storage_extended)
 
 
-@pytest.mark.uncollect('No IPA servers currently available')
-def test_appliance_console_ipa(ipa_creds, configured_appliance):
-    """'ap' launches appliance_console, '' clears info screen, '11' setup IPA, 'y' confirm setup
+def test_appliance_console_ipa(ipa_crud, configured_appliance):
+    """'ap' launches appliance_console, '' clears info screen, '11' setup IPA,
     + wait 40 secs and '' finish."""
 
-    command_set = ('ap', '', '11', ipa_creds['hostname'], ipa_creds['domain'], '',
-        ipa_creds['username'], ipa_creds['password'], TimedCommand('y', 40), '')
+    setup_ipa = '11'
+    command_set = ('ap', CLEAR, setup_ipa,
+                   ipa_crud.host1,
+                   ipa_crud.ipadomain or CLEAR,
+                   ipa_crud.iparealm or CLEAR,
+                   ipa_crud.ipaprincipal or CLEAR,
+                   ipa_crud.bind_password,
+                   TimedCommand('y', 40), CLEAR)
     configured_appliance.appliance_console.run_commands(command_set)
+    configured_appliance.sssd.wait_for_running()
+    assert configured_appliance.ssh_client.run_command("cat /etc/ipa/default.conf |"
+                                                       "grep 'enable_ra = True'")
 
-    def is_sssd_running(configured_appliance):
-        assert configured_appliance.ssh_client.run_command("systemctl status sssd | grep running")
-    wait_for(is_sssd_running, func_args=[configured_appliance])
-    return_code, output = configured_appliance.ssh_client.run_command(
-        "cat /etc/ipa/default.conf | grep 'enable_ra = True'")
-    assert return_code == 0
+    # Unconfigure to cleanup
+    # When setup_ipa option selected, will prompt to unconfigure, then to proceed with new config
+    command_set = ('ap', CLEAR, setup_ipa, TimedCommand('y', 40), TimedCommand('n', 5))
+    configured_appliance.appliance_console.run_commands(command_set)
```

Comment:
**QUESTION:** Shouldn't we assert on the return of the command being successful?

Reply:
Not possible in Appliance.ApplianceConsole.run_commands, unfortunately.